### ISSUE TITLE
Unexpected exit code 0

### ISSUE THREAD

**Issue by erikerikson** [hashicorp/terraform#25894]
TL;DR: both `terraform workspace select` and `terraform workspace new` fail to select/create a workspace but exit with code 0

### Terraform Version

```
Terraform v0.12.26
then
Terraform v0.13.0
```

### Terraform Configuration Files

```terraform
terraform {
  backend "azurerm" {
    key = "..." # a duplicate of locals.scope above.  Ensure these remain synchronized.
  }
}

provider "azurerm" {
  version = "=2.13.0"
  features {}
}

provider "null" {
  version = "~>2.1"
}
```

### Debug Output

Prior to upgrade (stopped mid change): // workspace "foo" does not exist
```
$ terraform workspace select foo
Error: Invalid expression

  on fm.tf line 383, in resource "azurerm_cosmosdb_mongo_collection" "...":
 383:
 384:

Expected the start of an expression, but found an invalid expression token.

C:\src\pivotal\ibms\fm-service>echo %errorlevel%
0
```
I fixed the error then tried again:
```
$ terraform workspace select foo

Workspace "foo" doesn't exist.

You can create this workspace with the "new" subcommand.

$ echo %errorlevel%
0
```
I turned on tracing and upgraded to the latest:
```
$ terraform workspace select foo

2020/08/17 23:25:14 [INFO] Terraform version: 0.13.0  
2020/08/17 23:25:14 [INFO] Go runtime version: go1.14.2
2020/08/17 23:25:14 [INFO] CLI args: []string{"C:\\...\\terraform.exe", "workspace", "select", "foo"}
2020/08/17 23:25:14 [DEBUG] Attempting to open CLI config file: C:\...\AppData\Roaming\terraform.rc
2020/08/17 23:25:14 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020/08/17 23:25:14 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020/08/17 23:25:14 [DEBUG] ignoring non-existing provider search directory C:\...\AppData\Roaming\terraform.d\plugins
2020/08/17 23:25:14 [DEBUG] ignoring non-existing provider search directory C:\...\AppData\Roaming\HashiCorp\Terraform\plugins
2020/08/17 23:25:14 [INFO] CLI command args: []string{"workspace", "select", "foo"}
2020/08/17 23:25:14 [TRACE] Meta.Backend: built configuration for "azurerm" backend with hash value 880177092
2020/08/17 23:25:14 [TRACE] Preserving existing state lineage "946f869f-8045-1992-5a28-ad35094a00fd"
2020/08/17 23:25:14 [TRACE] Preserving existing state lineage "946f869f-8045-1992-5a28-ad35094a00fd"
2020/08/17 23:25:14 [TRACE] Meta.Backend: working directory was previously initialized for "azurerm" backend
2020/08/17 23:25:14 [TRACE] backendConfigNeedsMigration: configuration values have changed, so migration is required
2020/08/17 23:25:14 [TRACE] Meta.Backend: backend configuration has changed (from type "azurerm" to type "azurerm")
Backend reinitialization required. Please run "terraform init".
Reason: Backend configuration changed for "azurerm"

The "backend" is the interface that Terraform uses to store state,
perform operations, etc. If this message is showing up, it means that the
Terraform configuration you're using is using a custom configuration for
the Terraform backend.

Changes to backend configurations require reinitialization. This allows
Terraform to setup the new configuration, copy existing state, etc. This is
only done during "terraform init". Please run that command now then try again.

If the change reason above is incorrect, please verify your configuration
hasn't changed and try again. At this point, no changes to your existing
configuration or state have been made.


Error: Initialization required. Please see the error message above.

$ echo %errorlevel%
0
```

I observe the same with new:
```
$ terraform workspace select <existing-workspace>

2020/08/17 23:43:07 [INFO] Terraform version: 0.13.0  
2020/08/17 23:43:07 [INFO] Go runtime version: go1.14.2
2020/08/17 23:43:07 [INFO] CLI args: []string{"C:\\...\\terraform.exe", "workspace", "new", "<existing-workspace>"}
2020/08/17 23:43:07 [DEBUG] Attempting to open CLI config file: C:\...\AppData\Roaming\terraform.rc
2020/08/17 23:43:07 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020/08/17 23:43:07 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020/08/17 23:43:07 [DEBUG] ignoring non-existing provider search directory C:\...\AppData\Roaming\terraform.d\plugins
2020/08/17 23:43:07 [DEBUG] ignoring non-existing provider search directory C:\...\AppData\Roaming\HashiCorp\Terraform\plugins
2020/08/17 23:43:07 [INFO] CLI command args: []string{"workspace", "new", "<existing-workspace>"}
2020/08/17 23:43:08 [TRACE] Meta.Backend: built configuration for "azurerm" backend with hash value 880177092
2020/08/17 23:43:08 [TRACE] Preserving existing state lineage "946f869f-8045-1992-5a28-ad35094a00fd"
2020/08/17 23:43:08 [TRACE] Preserving existing state lineage "946f869f-8045-1992-5a28-ad35094a00fd"
2020/08/17 23:43:08 [TRACE] Meta.Backend: working directory was previously initialized for "azurerm" backend
2020/08/17 23:43:08 [TRACE] backendConfigNeedsMigration: configuration values have changed, so migration is required
2020/08/17 23:43:08 [TRACE] Meta.Backend: backend configuration has changed (from type "azurerm" to type "azurerm")
Backend reinitialization required. Please run "terraform init".
Reason: Backend configuration changed for "azurerm"

The "backend" is the interface that Terraform uses to store state,
perform operations, etc. If this message is showing up, it means that the
Terraform configuration you're using is using a custom configuration for
the Terraform backend.

Changes to backend configurations require reinitialization. This allows
Terraform to setup the new configuration, copy existing state, etc. This is
only done during "terraform init". Please run that command now then try again.

If the change reason above is incorrect, please verify your configuration
hasn't changed and try again. At this point, no changes to your existing
configuration or state have been made.


Error: Initialization required. Please see the error message above.



$echo %errorlevel%
0
```

I could run init but the clear pattern is a total failure to succeed at running the commands (in a variety of ways) but I always got an exit code of 0.  The relevance is that I wanted to attempt to select a workspace and fall back to creating it (really I want an idempotent workplace creation) but doing something like `select || new` "succeeds" on the first clause (even though it didn't have the specified effect) and therefore fails to ensure the existence of the workspace.  Either I'm missing something dumb or this wasted too much of my time and doesn't meet long standing standards.  I'm confused that I'm reporting this.

I get around this by just running select and then new but it seems pretty wrong, thus this bug.

### Expected Behavior

I expected non-zero exit codes.

### Actual Behavior

Exit code is zero, commands not signalling failure as expected.

### Steps to Reproduce

1. `terraform workspace select adslkfjaldj`
2. `echo %errorlevel%`

### Additional Context

not that I know.  this happens consistently locally and in CI systems

**Comment by mildwonkey** [comment]
Hi @erikerikson , thank you for reporting this issue! Can you let me know what operating system and version you are using? We tried to reproduce this on Windows and have not been able to:

This example is with the non-existent workspace *only* (no "backend init" message"):
```
C:\Users\User\Downloads\long-filename-bug>.\terraform.exe workspace select doesnotexist
Workspace "doesnotexist" doesn't exist.
You can create this workspace with the "new" subcommand.
C:\Users\User\Downloads\long-filename-bug>echo %errorlevel%
1
```

I do see that you are also getting a "reinitialization required." message when you run the `workspace select` command. That's still expected to return an error code, and again when we coerce that situation on a windows machine we get the same result, an error: 

![image (1)](https://user-images.githubusercontent.com/6210214/90658668-d8551f00-e211-11ea-85fb-b633fa0a2b10.png)


FWIW we also can't reproduce the initial errorlevel=0 with the syntax error either, so I'm really not sure what's happening, but hopefully we can work together to reproduce this.

**Comment by erikerikson** [comment]
Thanks @mildmonkey.

I'm so sorry.  Thank you for attempting to reproduce.  In the process of trying to get a cleaner reproduction, I found my error.  I had misunderstood some docs.  There is no bug with Terraform here.

**Comment by mildwonkey** [comment]
No worries @erikerikson , thank you for the update 🎉
